docs/Python-API.rst

```diff
@@ -12,6 +12,7 @@ Data Structure API
     Dataset
```

Comment:
Please mention `Sequence` here:
https://github.com/microsoft/LightGBM/blob/master/docs/Python-Intro.rst#data-interface

Reply:
I've add some doc. Please take a look to see if this is appropriate.